recipes-support/snapd/snapd-go.inc

```diff
@@ -0,0 +1,64 @@
+inherit go-mod
+
+GO_IMPORT = "github.com/snapcore/snapd"
+
+# disable shared runtime for x86
+# https://forum.snapcraft.io/t/yocto-rocko-core-snap-panic/3261
+# GO_DYNLINK is set with arch overrides in goarch.bbclass
```

Comment:
Btw, I was doing the same thing for sysota but later reverted back to (mostly) defaults as thing got better in the go ecosystem. I think most of those entries are pointless now. I'd keep them and transition out of them over time.